### PR TITLE
Add initial port of Node.js Compat Streams

### PR DESCRIPTION
**Pull request by jasnell**
While there is still a lot to do here to finish up adding tests, adding the Web streams adapters, and implementing the utility consumers, this PR is already rather large at 7700+ lines changed. It's still gated as experimental so I'll iterate on this to take care of the remaining todos in a separate PR.

Derived from: https://github.com/nodejs/readable-stream

TODO:
* [ ] Tests, Tests, and more Tests
* [ ] Node.js streams <=> Web streams adapters (https://nodejs.org/dist/latest-v19.x/docs/api/stream.html#streamreadabletowebstreamreadable-options)
* [ ] Utility consumers (https://nodejs.org/dist/latest-v19.x/docs/api/webstreams.html#utility-consumers)
* [ ] Documentation
* [x] Samples

### CODE REVIEW

File: src/workerd/jsg/modules.c++ [hunk 1]
```diff
@@ -74,11 +74,11 @@ v8::MaybeLocal<v8::Module> resolveCallback(v8::Local<v8::Context> context,
         "No such module \"", targetPath.toString(),
         "\".\n  imported from \"", ref.specifier.toString(), "\"")
         .module.getHandle(js);
-
-  })) {
-    isolate->ThrowException(makeInternalError(isolate, kj::mv(*exception)));
+  }, [&](Value value) {
+KJ_DBG(value.getHandle(js));
```
Comment (mikea) [inline src/workerd/jsg/modules.c++:78]: cleanup?

File: src/workerd/jsg/modules.c++ [hunk 2]
```diff
@@ -279,9 +279,9 @@ void instantiateModule(jsg::Lock& js, v8::Local<v8::Module>& module) {
   KJ_ASSERT(!module.IsEmpty());
   auto isolate = js.v8Isolate;
   auto context = isolate->GetCurrentContext();
+  js.tryCatch([&] {
```
Comment (mikea) [inline src/workerd/jsg/modules.c++:282]: nit: indent below. Or did you intend to remove this tryCatch?

**Review by mikea — APPROVED** [review]
LG with couple nits in C++ code.

I didn't really review .js since I suppose it comes mostly from node.

File: src/workerd/jsg/modules.c++ (hunk 1 above)
Comment (jasnell) [reply src/workerd/jsg/modules.c++:78]: doh!

File: src/workerd/jsg/modules.c++ (hunk 2 above)
Comment (jasnell) [reply src/workerd/jsg/modules.c++:282]: intended to remove. that was missed cleanup

File: src/node/_stream_duplex.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
```
[the remark this answers is not in the document]
Comment (jasnell) [reply src/node/_stream_duplex.js:1]: Many many years ago, before we added the hidden `lib/internal` folder in node.js, they used a convention that `_` prefixed modules were "private" even though users could still require them. It was silly and dumb but it turns out there is code in the wild that actually imports these.

File: src/workerd/jsg/modules.c++ [hunk 4]
```diff
@@ -298,6 +298,9 @@ void instantiateModule(jsg::Lock& js, v8::Local<v8::Module>& module) {
     case v8::Promise::kFulfilled:
     break;
   }
+  }, [&](Value exception) {
+    KJ_DBG(exception.getHandle(js));
```
[the remark this answers is not in the document]
Comment (jasnell) [reply src/workerd/jsg/modules.c++]: it was meant to be removed.

**Pull request merged**